src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -15,7 +16,15 @@ public class WebSocketsTransport : ITransport
         private ClientWebSocket _webSocket = new ClientWebSocket();
         private IChannelConnection<Message> _application;
         private CancellationToken _cancellationToken = new CancellationToken();
+        private readonly ILogger _logger;
 
+        public WebSocketsTransport()
+        {
+        }
+        public WebSocketsTransport(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<WebSocketsTransport>();
```

Comment:
Consider using NullLoggerFactory if nothing is passed. This way we will not run into NREs because someone forgot to check for null.